### PR TITLE
Feature: Top Traders - Top 20

### PR DESCRIPTION
**Pull request by ChefHutch**
Expand the Top Traders section to show the next 15 traders on `ExpandableLabel` click

- Between **576 - 852px** (where the card has the full-screen) and again above **1080px**, the expanded wrapper shows as a three-column grid.
- Above & below those values, it renders as a block element.
- Resulting in lots of media queries to make this work across screen sizes.

Preview https://elegant-payne-0f0170.netlify.app/competition

### Below 576 - 1 column
Screenshot 2021-04-08 at 13 25 54

### 576 - 852px - 3 column
Screenshot 2021-04-08 at 12 33 42

### 852 - 1080 - 1 column
Screenshot 2021-04-08 at 12 33 59

### Above 1080 - 3 column
Screenshot 2021-04-08 at 12 34 10

### CODE REVIEW

File: src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    margin-right: 16px;
+  }
+`
+
+const GridItem: React.FC<{ traderData?: LeaderboardDataItem }> = ({ traderData }) => {
+  const { address, volume, teamId, rank } = traderData
+
+  const icon = {
```
Comment (RabbitDoge) [inline src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx:74]: why don't you use an array ?

Comment (ChefHutch) [reply src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx]: Did this & moved to the level above as it's shared by both components 👌

File: src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  margin-left: 4px;
+
+  ${({ theme }) => theme.mediaQueries.xs} {
+    margin-left: 8px;
+  }
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    margin-left: 16px;
+  }
+`
```
Comment (hachiojidev) [inline src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx:47]: Use responsive margin prop?

File: src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+const TeamImageWrapper = styled(Flex)`
+  ${({ theme }) => theme.mediaQueries.xl} {
+    position: absolute;
+    right: 0;
+    bottom: 0;
+  }
+`
+
+const GridItem: React.FC<{ traderData?: LeaderboardDataItem; teamImages?: React.ReactNode[] }> = ({
```
Comment (hachiojidev) [inline src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx:95]: I have noticed that you frequently make props optional. In general if a prop is optional then there should be code in your component dealing with the cases that is comes down undefined. In both cases I think this component will throw an error/look weird if a prop comes down as `undefined`.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx (hunk 2 above)
Comment (ChefHutch) [reply src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx:47]: 👍

File: src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx (hunk 3 above)
Comment (ChefHutch) [reply src/views/TradingCompetition/components/TeamRanks/TopTradersCard/ExpandedGridItem.tsx]: Thanks for this Hachi - I opted to leave that traderData optional but with some sensible defaults. TeamImages is required, passed down via static imports in this case.

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**